Visual Design Cleanup for Headers - Filters and header buttons

akiselev opened the pull request:
See the [notion doc](https://www.notion.so/metabase/Visual-Design-Cleanup-for-Headers-3475a1338cb840f28904063553b8ed87) and #19909 

**Repro Steps**
* Create a question
* Sample Dataset > Product
* Add a filter Price less than 30


**Before**
![image](https://user-images.githubusercontent.com/653604/150985735-4dd953cc-10ab-476a-9069-f22f9fb26273.png)

![image](https://user-images.githubusercontent.com/653604/150986041-57e558b0-7e69-46aa-a538-06a843eccbeb.png)


**After**
![image](https://user-images.githubusercontent.com/653604/150985989-a45d2f6c-9015-4de9-9902-86063fe56213.png)

![image](https://user-images.githubusercontent.com/653604/150986119-51ba1d7c-cd33-48cd-8429-e6775481c023.png)

Commit:
header filter changes as separate PR

Commit:
add span around count and fix filter bar padding

Commit:
fix broken intermittent test

gusaiani reviewed (approved):
Code looks pretty good and so does the UI.

One tweak I'd suggest: optically centering the text in the Filter and Summarize buttons.

padding-top: 12px;
padding-bottom: 14px;

seem to do it.

![image](https://user-images.githubusercontent.com/380816/151403717-205ac30c-fd45-4193-9e14-98ff2079b708.png)

kulyk commented:
Looks cool! Noticed that "Filter" and "Summarize" buttons loose the hover outline after you open and then close them, see:

![CleanShot 2022-01-27 at 19 11 46](https://user-images.githubusercontent.com/17258145/151408912-9ed29d38-6585-4faf-9fb6-fa76faac97b7.gif)

Commit:
another fix attempt

Commit:
fix button hover CSS

Commit:
attempt another fix for #16386 repro

Commit:
another attempt

Commit:
trying again and again

Commit:
timeout fix

Commit:
spinning wheels

ariya commented on frontend/src/metabase/query_builder/components/view/ViewHeader.styled.jsx:
```diff
@@ -53,3 +54,38 @@ export const DatasetHeaderButtonContainer = styled.div`
+      active ? alpha(color, 0.8) : "transparent"};
+    color: ${({ active }) => (active ? "white" : color("text-dark"))};
+  }
+  transition: background 300ms linear, border 300ms linear;
```
While this is likely subtle enough, doesn't hurt to support reduced motion (see #18169).

akiselev replied:
Nice catch! Fixed

Commit:
ridiculous e2e failure

Commit:
cellData based fix

Commit:
added prefers-reduced-motion rule for header buttons

akiselev commented:
> Looks cool! Noticed that "Filter" and "Summarize" buttons loose the hover outline after you open and then close them, see:
> 
> ![CleanShot 2022-01-27 at 19 11 46](https://user-images.githubusercontent.com/17258145/151408912-9ed29d38-6585-4faf-9fb6-fa76faac97b7.gif)

Fixed

> Code looks pretty good and so does the UI.
> 
> One tweak I'd suggest: optically centering the text in the Filter and Summarize buttons.
> 
> padding-top: 12px; padding-bottom: 14px;
> 
> seem to do it.
> 
> ![image](https://user-images.githubusercontent.com/380816/151403717-205ac30c-fd45-4193-9e14-98ff2079b708.png)

Added to the epic as a future update
